test/cases/api.sh

```diff
@@ -0,0 +1,169 @@
+# terminates in any way.
+#
+
+WORKDIR=$(mktemp -d)
+
+AWS_REGION=${AWS_REGION-}
+AWS_BUCKET=${AWS_BUCKET-}
+AWS_ACCESS_KEY_ID=${AWS_ACCESS_KEY_ID-}
+AWS_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY-}
+AWS_API_TEST_SHARE_ACCOUNT=${AWS_API_TEST_SHARE_ACCOUNT-}
+
+[[ "${1-}" == '-debug' ]] || {
+  # Verify AWS env vars exist
+  printenv AWS_REGION AWS_BUCKET AWS_ACCESS_KEY_ID AWS_SECRET_ACCESS_KEY AWS_API_TEST_SHARE_ACCOUNT > /dev/null
+}
+
+#
+# Make sure /openapi.json and /version endpoints return success
+#
+
+# org_id 000000
+AuthString0="eyJlbnRpdGxlbWVudHMiOnsiaW5zaWdodHMiOnsiaXNfZW50aXRsZWQiOnRydWV9LCJzbWFydF9tYW5hZ2VtZW50Ijp7ImlzX2VudGl0bGVkIjp0cnVlfSwib3BlbnNoaWZ0Ijp7ImlzX2VudGl0bGVkIjp0cnVlfSwiaHlicmlkIjp7ImlzX2VudGl0bGVkIjp0cnVlfSwibWlncmF0aW9ucyI6eyJpc19lbnRpdGxlZCI6dHJ1ZX0sImFuc2libGUiOnsiaXNfZW50aXRsZWQiOnRydWV9fSwiaWRlbnRpdHkiOnsiYWNjb3VudF9udW1iZXIiOiIwMDAwMDAiLCJ0eXBlIjoiVXNlciIsInVzZXIiOnsidXNlcm5hbWUiOiJ1c2VyIiwiZW1haWwiOiJ1c2VyQHVzZXIudXNlciIsImZpcnN0X25hbWUiOiJ1c2VyIiwibGFzdF9uYW1lIjoidXNlciIsImlzX2FjdGl2ZSI6dHJ1ZSwiaXNfb3JnX2FkbWluIjp0cnVlLCJpc19pbnRlcm5hbCI6dHJ1ZSwibG9jYWxlIjoiZW4tVVMifSwiaW50ZXJuYWwiOnsib3JnX2lkIjoiMDAwMDAwIn19fQ=="
+
+# org_id 000001 (wrong org_id)
+AuthString1="eyJlbnRpdGxlbWVudHMiOnsiaW5zaWdodHMiOnsiaXNfZW50aXRsZWQiOnRydWV9LCJzbWFydF9tYW5hZ2VtZW50Ijp7ImlzX2VudGl0bGVkIjp0cnVlfSwib3BlbnNoaWZ0Ijp7ImlzX2VudGl0bGVkIjp0cnVlfSwiaHlicmlkIjp7ImlzX2VudGl0bGVkIjp0cnVlfSwibWlncmF0aW9ucyI6eyJpc19lbnRpdGxlZCI6dHJ1ZX0sImFuc2libGUiOnsiaXNfZW50aXRsZWQiOnRydWV9fSwiaWRlbnRpdHkiOnsiYWNjb3VudF9udW1iZXIiOiIwMDAwMDAiLCJ0eXBlIjoiVXNlciIsInVzZXIiOnsidXNlcm5hbWUiOiJ1c2VyIiwiZW1haWwiOiJ1c2VyQHVzZXIudXNlciIsImZpcnN0X25hbWUiOiJ1c2VyIiwibGFzdF9uYW1lIjoidXNlciIsImlzX2FjdGl2ZSI6dHJ1ZSwiaXNfb3JnX2FkbWluIjp0cnVlLCJpc19pbnRlcm5hbCI6dHJ1ZSwibG9jYWxlIjoiZW4tVVMifSwiaW50ZXJuYWwiOnsib3JnX2lkIjoiMDAwMDAxIn19fQ=="
+
+CurlCmd='curl -s -w %{http_code}'
+Header="x-rh-identity: $AuthString0"
+Address="localhost"
+Port="8086"
+Version="1.0"
+MajorVersion="1"
+BaseUrl="http://$Address:$Port/api/image-builder/v$Version"
+BaseUrlMajorVersion="http://$Address:$Port/api/image-builder/v$MajorVersion"
+
+function GetResponse() {
+  arr=($1)
+  echo "${arr[@]::${#arr[@]}-1}"
+}
+
+function GetExitCode() {
+  arr=($1)
+  echo "${arr[-1]}"
+}
+
+### Case: Get version
+function Test_GetVersion() {
+  result=$($CurlCmd -H "$Header" "$url/version")
+  ver="$(GetResponse "$result" | jq .version -r)"
+  [[ $ver == "$Version" ]]
+  exit_code=$(GetExitCode "$result")
+  [[ $exit_code == 200 ]]
+}
+
+### Case: Get openapi.json
+function Test_GetOpenapi() {
+  result=$($CurlCmd -H "$Header" "$url/openapi.json")
+  exit_code=$(GetExitCode "$result")
+  [[ $exit_code == 200 ]]
+}
+
+
+### Case: Post to composer
+function Test_PostToComposer() {
+  #
+  # Prepare a request to be sent to the composer API.
+  #
+
+  REQUEST_FILE="${WORKDIR}/request.json"
+  ARCH=$(uname -m)
+  SNAPSHOT_NAME=$(uuidgen)
+
+  case $(set +x; . /etc/os-release; echo "$ID-$VERSION_ID") in
```

Comment:
Currently we only test on `rhel-8` and `fedora-32` in image-builder. So f33, rhel8.4 and centos aren't used atm.

Reply:
Thanks! Only keep rhel-8(8.2/8.3/8.4) and fedora-32 here.